tests/rptest/tests/e2e_shadow_indexing_test.py

```diff
@@ -88,6 +91,103 @@ def tearDown(self):
 
 
 class EndToEndShadowIndexingTest(EndToEndShadowIndexingBase):
+    def _all_uploads_done(self):
+        topic_description = self.rpk.describe_topic(self.topic)
+        partition = next(topic_description)
+
+        hwm = partition.high_watermark
+
+        manifest = None
+        try:
+            bucket = BucketView(self.redpanda)
+            manifest = bucket.manifest_for_ntp(self.topic, partition.id)
+        except Exception as e:
+            self.logger.info(
+                f"Exception thrown while retrieving the manifest: {e}")
+            return False
+
+        top_segment = max(manifest['segments'].values(),
+                          key=lambda seg: seg['base_offset'])
+        uploaded_raft_offset = top_segment['committed_offset']
+        uploaded_kafka_offset = uploaded_raft_offset - top_segment[
+            'delta_offset_end']
+        self.logger.info(
+            f"Remote HWM {uploaded_kafka_offset} (raft {uploaded_raft_offset}), local hwm {hwm}"
+        )
+
+        # -1 because uploaded offset is inclusive, hwm is exclusive
+        if uploaded_kafka_offset < (hwm - 1):
+            return False
+
+        return True
+
+    @cluster(num_nodes=7)
```

Comment:
protip: you can use fewer nodes if you .free() the producer/consumer objects each time you're done with them.

This is also better because it means the test will end up re-using the same node that has the valid_offsets.json file from the producer, when running the consumer.